src/components/ui/Modal/Modal.tsx

```diff
@@ -12,8 +12,10 @@ export type ModalChildrenProps = {
   fadeIn: () => void
 }
 
+type ModalChildrenFunction = (props: ModalChildrenProps) => ReactNode
+
 export type ModalProps = Omit<UIModalProps, 'isOpen' | 'children'> & {
-  children: (props: ModalChildrenProps) => ReactNode | ReactNode
```

Comment:
Curious if the [mentioned duplication](https://user-images.githubusercontent.com/11325562/178339767-65387f77-1239-4660-8d55-fb6d27b7c410.png) is because SonarQube thinks `… ReactNode | ReactNode` is the same thing?

It looks duplicated but we know it's not, it's a _function (that returns a ReactNode) **OR** a ReactNode_. Adding `()` may be another alternative to this (e.g. `((props: ModalChildrenProps) => ReactNode) | ReactNode`), but I like your solution better!

Reply:
I think so. SonarQube seems to confuse it. Adding `()` should work as well 😉 .